platforms/core-configuration/model-core/src/main/java/org/gradle/api/internal/provider/ValueSanitizers.java

```diff
@@ -50,40 +66,44 @@ public Object sanitize(@Nullable Object value) {
     private static final ValueSanitizer<Object> IDENTITY_SANITIZER = new ValueSanitizer<Object>() {
         @Override
         @Nullable
-        public Object sanitize(@Nullable Object value) {
+        public Object sanitize(Class<?> valueType, @Nullable Object value) {
             return value;
         }
     };
 
-    private static final ValueCollector<Object> IDENTITY_VALUE_COLLECTOR = valueCollectorWithValueSanitizer(IDENTITY_SANITIZER);
-    private static final ValueCollector<Object> STRING_VALUE_COLLECTOR = valueCollectorWithValueSanitizer(STRING_VALUE_SANITIZER);
-    private static final ValueCollector<Object> LONG_VALUE_COLLECTOR = valueCollectorWithValueSanitizer(LONG_VALUE_SANITIZER);
+    private static final ValueCollector<Object> IDENTITY_VALUE_COLLECTOR = valueCollectorWithValueSanitizer(Object.class, IDENTITY_SANITIZER);
+    private static final ValueCollector<Object> STRING_VALUE_COLLECTOR = valueCollectorWithValueSanitizer(String.class, STRING_VALUE_SANITIZER);
+    private static final ValueCollector<Object> LONG_VALUE_COLLECTOR = valueCollectorWithValueSanitizer(Long.class, LONG_VALUE_SANITIZER);
 
-    public static <T> ValueSanitizer<T> forType(Class<? extends T> targetType) {
+    public static <T> ValueSanitizer<T> forType(@Nullable Class<? extends T> targetType) {
         if (String.class.equals(targetType)) {
             return Cast.uncheckedCast(STRING_VALUE_SANITIZER);
         } else if (Long.class.equals(targetType)) {
             return Cast.uncheckedCast(LONG_VALUE_SANITIZER);
+        } else if (targetType != null && targetType.isEnum()) {
+            return Cast.uncheckedCast(ENUM_VALUE_SANITIZER);
         } else {
             return Cast.uncheckedCast(IDENTITY_SANITIZER);
         }
     }
 
-    public static <T> ValueCollector<T> collectorFor(Class<? extends T> elementType) {
+    public static <T> ValueCollector<T> collectorFor(@Nullable Class<? extends T> elementType) {
         if (String.class.equals(elementType)) {
             return Cast.uncheckedCast(STRING_VALUE_COLLECTOR);
         } else if (Long.class.equals(elementType)) {
             return Cast.uncheckedCast(LONG_VALUE_COLLECTOR);
+        } else if (elementType != null && elementType.isEnum()) {
```

Comment:
❓ It looks like you cannot invoke string-to-enum conversion with lists or maps in plain groovy, so we probably don't need it here.

Reply:
It seems we support for it somehow. An usecase can also be found in the documentation. See here:
https://github.com/gradle/gradle/blob/497662cfed81f58b000b0c49b0946fd83e1ed4fa/platforms/software/testing-base/src/main/java/org/gradle/api/tasks/testing/logging/TestLoggingContainer.java#L46

I also checked with this test https://github.com/gradle/gradle/blob/497662cfed81f58b000b0c49b0946fd83e1ed4fa/platforms/jvm/testing-jvm/src/integTest/groovy/org/gradle/testing/junit/AbstractJUnitConsoleLoggingIntegrationTest.groovy#L117-L172
by changing `events "standardOut", "standardError"` to `events = ["standardOut", "standardError"]` and it just works

Reply:
>It seems we support it somehow. An usecase can also be found in the documentation

It looks like TestLogging does that in an ad-hoc way though.